rossml/elements.py

```diff
@@ -0,0 +1,687 @@
+
+    Before setting data to the neural network, check what are the required features
+    according to the loaded model.
+
+    First, load the neural network:
+
+    >>> model = rsml.Model(nn_model)
+
+    Check for the features:
+    >>> features = model.features
+
+    Now, enter the respective values for each key in "model.features". Or use other
+    classes with preset parameters.
+
+    >>> seal = rsml.AnnBearingElement(
+    ...     n=0,
+    ...     arq=nn_model,
+    ...     seal_radius=141.61,
+    ...     number_of_teeth=22,
+    ...     tooth_pitch=8.58,
+    ...     tooth_height=8.37,
+    ...     radial_clearance=0.16309,
+    ...     ethane=0.14548,
+    ...     propane=0.00542,
+    ...     isobutan=.17441,
+    ...     butane=0.22031,
+    ...     nitrogen=0.10908,
+    ...     methane=0.11907,
+    ...     hydrogen=0.11584,
+    ...     oxygen=0.05794,
+    ...     co2=0.05224,
+    ...     reservoir_temperature=25.0,
+    ...     reservoir_pressure=568.90,
+    ...     sump_pressure=5.3,
+    ...     inlet_tangential_velocity_ratio=0.617,
+    ...     whirl_speed=8310.5,
+    ...     speeds=[7658.3],
+    ... )
+    >>> seal # doctest: +ELLIPSIS
+    AnnBearingElement(n=0...
+
+    If kwargs has different argumments than model.features, an error is raised
+    informing how many which are the features and how many kwargs has been entered.
+    Users can copy the list directly from the error message to set the correct keys in
+    kwargs.
+
+    >>> seal = rsml.AnnBearingElement(
+    ...     n=0,
+    ...     arq=nn_model,
+    ...     seal_radius=141.61,
+    ...     number_of_teeth=22,
+    ...     tooth_pitch=8.58,
+    ...     tooth_height=8.37,
+    ...     radial_clearance=0.16309,
+    ...     methane=0.11907,
+    ...     hydrogen=0.11584,
+    ...     oxygen=0.05794,
+    ...     co2=0.05224,
+    ...     reservoir_temperature=25.0,
+    ...     reservoir_pressure=568.90,
+    ...     sump_pressure=5.3,
+    ...     inlet_tangential_velocity_ratio=0.617,
+    ...     whirl_speed=8310.5,
+    ...     speeds=[7658.3],
+    ... ) # doctest: +ELLIPSIS
+    KeyError...
+    """
+
+    def __init__(self, arq=None, n=None, n_link=None, scale_factor=1.0, **kwargs):
+        # loading neural network model
+        model = Model(arq)
+        features = model.features
+        reordered_dict = {}
+
+        # checking data consistency
+        if any(key not in kwargs.keys() for key in features):
+            raise KeyError(
+                f"Model '{arq}' has the following {len(list(features))} features: "
+                f"{list(features)}, and {len(kwargs)} are given. "
+                f"Check the **kwargs dictionary for the same keys."
+            )
+
+        size = len(kwargs["speeds"])
+
+        if any(len(v) != size for k, v in kwargs.items() if isinstance(v, Iterable)):
+            raise ValueError(
+                "Some keyword arguments does not have the same length than 'speeds'."
+            )
```

Comment:
Does this mean that if we have a list of `speeds=[100, 200, ...]` we will also need a list for the other arguments? Like `sump_pressure=[...]`?
If this is the case, I would recommend changing this part of the code. 
Our use case is that we provide only one value for the process parameters such as temperature and pressures, and a list/range of speeds for the component.

Reply:
> Does this mean that if we have a list of `speeds=[100, 200, ...]` we will also need a list for the other arguments? Like `sump_pressure=[...]`

No, you can enter single values for each argument and a list of speed values to `speeds`. Example: single values to every argument and a list with 3 speed values to `speeds`. You can see the class return a BearingElement with speed depedent coefficients (3 values per coef. and 3 values in frequency).
```python
>>> import rossml as rsml
>>> rsml.available_models()
>>> nn_model = "test_model"
>>> model = rsml.Model(nn_model)
>>> seal = rsml.AnnBearingElement(
...     n=0,
...     arq=nn_model,
...     seal_radius=141.61,
...     number_of_teeth=22,
...     tooth_pitch=8.58,
...     tooth_height=8.37,
...     radial_clearance=0.16309,
...     ethane=0.14548,
...     propane=0.00542,
...     isobutan=.17441,
...     butane=0.22031,
...     nitrogen=0.10908,
...     methane=0.11907,
...     hydrogen=0.11584,
...     oxygen=0.05794,
...     co2=0.05224,
...     reservoir_temperature=25.0,
...     reservoir_pressure=568.90,
...     sump_pressure=5.3,
...     inlet_tangential_velocity_ratio=0.617,
...     whirl_speed=8310.5,
...     speeds=[1000, 2000, 3000],
... )
>>> seal
AnnBearingElement(n=0, n_link=None,
 kxx=array([-1670677.125, -1694202.625, -1683276.625]), kxy=array([ 94719.9375  , 126924.359375, 194036.828125]),
 kyx=array([-106105.765625, -144189.40625 , -206884.875   ]), kyy=array([-1668276.875, -1692516.5  , -1704435.5  ]),
 cxx=array([500.61611938, 527.29345703, 572.21862793]), cxy=array([-436.73156738, -564.35974121, -743.44049072]),
 cyx=array([-436.73156738, -564.35974121, -743.44049072]), cyy=array([479.01638794, 499.57983398, 536.6864624 ]),
 frequency=[1000. 2000. 3000.], tag='test_model')
```

Actually you could do both ways. One, like I just said, and the other passing lists to every argument (in this case, the lists size should match the `speeds` list size.

Maybe I should change the error message to inform when inputting lists to other arguments rather than `speeds` theirs size should match the `speeds` size.